R/utils.R

```diff
@@ -551,7 +551,15 @@ verify_attr <- function(proposed, schema, layoutAttr = FALSE) {
     
     # do the same for "sub-attributes"
     if (identical(role, "object") && is.recursive(proposed[[attr]])) {
-      proposed[[attr]] <- verify_attr(proposed[[attr]], attrSchema, layoutAttr = layoutAttr)
+      # The "dimensions" attribute requires a special treatment as 
+      # it is an unnamed list and hence will be skipped by `for (attr in names(proposed))
+      if (attr == "dimensions") {
```

Comment:
Seems there is actually a more general problem here that's beyond just `dimensions`. Some attributes say `role: object`, but are actually expecting a list of objects. Here is the schema for both `dimensions` and `transforms`:

<img width="483" alt="Screenshot 2024-09-03 at 5 10 03 PM" src="https://github.com/user-attachments/assets/54b4a9d3-9c59-471e-8ae7-413e7ae1d17e">

<img width="483" alt="Screenshot 2024-09-03 at 5 10 03 PM" src="https://github.com/user-attachments/assets/96b880ec-6d00-4ff0-bd15-8c8e6a5612a8">


So, I think this code should be closer to:

```r
if (identical(role, "object") && is.recursive(proposed[[attr]])) {
      # Some attributes (e.g., dimensions, transforms) are actually 
      # a list of objects (even though they, confusingly, have role: object)
      # In those cases, we actually want to verify each list element
      attr_ <- sub("s$", "", attr)
      is_list_attr <- ("items" %in% names(attrSchema)) && 
        (attr_ %in% names(attrSchema$items))
      if (is_list_attr) {
        proposed[[attr]] <- lapply(proposed[[attr]], function(x) {
          verify_attr(x, attrSchema$items[[attr_]])
        })
      } else {
        proposed[[attr]] <- verify_attr(proposed[[attr]], attrSchema, layoutAttr = layoutAttr)
      }
    }
```

Reply:
Thx for the review and the suggestions. Just added and comitted.